### PR TITLE
fix for issue where $brooklyn:external refs in provisioning.properties are resolved too early

### PR DESCRIPTION
**Pull request by ahgittin**
several more tests, including one which failed before the fix -- and of course the fix itself

### CODE REVIEW

File: camp/camp-brooklyn/src/test/java/org/apache/brooklyn/camp/brooklyn/LocationExternalConfigYamlTest.java [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import com.google.common.base.Joiner;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterables;
+
+public class LocationExternalConfigYamlTest extends AbstractYamlRebindTest {
+
+	private static final Logger log = LoggerFactory.getLogger(ExternalConfigYamlTest.class);
```
Comment (nakomis) [inline camp/camp-brooklyn/src/test/java/org/apache/brooklyn/camp/brooklyn/LocationExternalConfigYamlTest.java]: Tab character...

File: camp/camp-brooklyn/src/test/java/org/apache/brooklyn/camp/brooklyn/LocationExternalConfigYamlTest.java [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+        Assert.assertTrue(rawConfig.get() instanceof DeferredSupplier, "Expected deferred raw value; got "+rawConfig.get());
+    }
+    
+    @Test(groups="Integration")
+    public void testProvisioningPropertyInheritance() throws Exception {
+        String yaml = Joiner.on("\n").join(
+                "services:",
+                "- type: "+EmptySoftwareProcess.class.getName(),
+                "  provisioning.properties:",
+                "      simple: 42",
```
Comment (nakomis) [inline camp/camp-brooklyn/src/test/java/org/apache/brooklyn/camp/brooklyn/LocationExternalConfigYamlTest.java]: Indentation should be 2 spaces, but doesn't really matter

File: camp/camp-brooklyn/src/test/java/org/apache/brooklyn/camp/brooklyn/LocationExternalConfigYamlTest.java [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+                "- type: "+EmptySoftwareProcess.class.getName(),
+                "location:",
+                "  localhost:",
+                "    my.config.key: $brooklyn:external(\"myprovider\", \"mykey\")");
+
+        origApp = (StartableApplication) createAndStartApplication(new StringReader(yaml));
+
+        Entity entity = Iterables.getOnlyElement( origApp.getChildren() );
+        Location l = Iterables.getOnlyElement( entity.getLocations() );
+        log.info("Location: "+l);
```
Comment (nakomis) [inline camp/camp-brooklyn/src/test/java/org/apache/brooklyn/camp/brooklyn/LocationExternalConfigYamlTest.java]: These log statements don't really give much info in the context of reading a log file, should be removed, or give more context

**Comment by ahgittin** [comment]
thx @nakomis, all addressed

**Pull request merged**